packages/flutter/lib/src/widgets/fade_in_image.dart

```diff
@@ -442,6 +475,20 @@ class _FadeInImageState extends State<FadeInImage> with TickerProviderStateMixin
       : _imageResolver._imageInfo;
   }
 
+  bool _flipHorizontally(BuildContext context) {
+    if (!widget.matchTextDirection)
+      return false;
+    final TextDirection textDirection = Directionality.of(context);
+    assert(textDirection != null, 'matchTextDirection can only be used when an ambient Directionality widget is in scope.');
+    switch (textDirection) {
+      case TextDirection.rtl:
+        return true;
+      case TextDirection.ltr:
+        return false;
+    }
+    return null;
+  }
```

Comment:
It's a bummer to have this code three times.  Maybe pull it out into a static function on `RawImage`?  That's probably better than a free `textDirectionIsReversed` function because folks will be less tempted to use it for RTL generally.

Reply:
It's a lot of words, but the actual code is trivial here. It's basically a very verbose ternary operator. Would you feel the same way if I wrote it as this?:

```dart
   assert(!widget.matchTextDirection || debugCheckHasDirectionality(context));
   return widget.matchTextDirection && Directionality.of(context) == TextDirection.rtl;
```

It's basically identical to this logic, I'm just avoiding writing that because it's an enum and I don't want to set the pattern of using direct comparisons on enums (as opposed to using switch). (Though it's unclear to me whether that's worth it for this case; we're not likely to add another text direction.)

Reply:
The reason not to use direct comparisons is that the enum really has three values (RTL, LTR, and null).  Even if you assert, in production you could have the null value.  When that happens, you want an exception of some sort instead of silently assuming LTR (or RTL).

Reply:
You don't necessarily get an exception in release mode even with the switch. For example, when I had the switch in the ImageDecoration code, it did the same thing for null and for LTR, which is to say, nothing, leaving it at the default (false).

Reply:
Yeah, I thought about that when reading the code, but it was below the threshold of things that I usually comment on.  :)

Reply:
Well in any case, I don't think we should optimize for getting exceptions when things are broken in release mode. If your code would assert in debug builds, then behaviour in release builds is undefined, IMHO.